e2e_test/mock.go

```
@@ -0,0 +1,98 @@
+		"headers": r.Header,
+		"query":   r.URL.Query(),
+	}
+
+	if r.Body != nil {
+		body, err := io.ReadAll(r.Body)
+		if err == nil {
+			response["body"] = string(body)
+		}
+	}
+
+	json.NewEncoder(w).Encode(response)
```

golangci-lint

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)